src/core/net/dnssd_server.hpp

```diff
@@ -188,19 +127,176 @@ class Server : public InstanceLocator, private NonCopyable
+    };
+
+    /**
+     * This constructor initializes the object.
+     *
+     * @param[in]  aInstance     A reference to the OpenThread instance.
+     *
+     */
+    explicit Server(Instance &aInstance);
+
+    /**
+     * This method starts the DNS-SD server.
+     *
+     * @retval kErrorNone     Successfully started the DNS-SD server.
+     * @retval kErrorFailed   If failed to open or bind the UDP socket.
+     *
+     */
+    Error Start(void);
+
+    /**
+     * This method stops the DNS-SD server.
+     *
+     */
+    void Stop(void);
+
+    /**
+     * This method sets DNS-SD query callbacks.
+     *
+     * @param[in] aSubscribe    A pointer to the callback function to subscribe a service or service instance.
+     * @param[in] aUnsubscribe  A pointer to the callback function to unsubscribe a service or service instance.
+     * @param[in] aContext      A pointer to the application-specific context.
+     *
+     */
+    void SetQueryCallbacks(otDnssdQuerySubscribeCallback   aSubscribe,
+                           otDnssdQueryUnsubscribeCallback aUnsubscribe,
+                           void *                          aContext);
+
+    /**
+     * This method notifies a discovered service instance.
+     *
+     * @param[in] aServiceFullName  The null-terminated full service name.
+     * @param[in] aInstanceInfo     A pointer to the discovered service instance information.
+     *
+     */
+    void HandleDiscoveredServiceInstance(const char *aServiceFullName, const otDnssdServiceInstanceInfo &aInstanceInfo);
+
+    /**
+     * This method notifies a discovered host.
+     *
+     * @param[in] aHostFullName     The null-terminated full host name.
+     * @param[in] aHostInfo         A pointer to the discovered host information.
+     *
+     */
+    void HandleDiscoveredHost(const char *aHostFullName, const otDnssdHostInfo &aHostInfo);
+
+    /**
+     * This function aquires the next query in the server.
+     *
+     * @param[in] aQuery            The query pointer. Pass NULL to get the first query.
+     *
+     * @returns  A pointer to the query or NULL if no more queries.
+     *
+     */
+    const QueryTransaction *GetNextQuery(const QueryTransaction *aQuery) const;
+
+    /**
+     * This function aquires the dns-sd query type and name for a specific query.
+     *
+     * @param[in]   aHeader           The query header.
+     * @param[in]   aMessage          The query message.
+     * @param[out]  aNameOutput       The name output buffer, recommended be at least 255 bytes long.
+     * @param[out]  aNameSize         The size of @p aNameOutput
+     *
+     * @returns The dns-sd query type.
+     *
+     */
+    static DnsQueryType GetQueryTypeAndName(const Header & aHeader,
+                                            const Message &aMessage,
+                                            char *         aNameOutput,
+                                            uint16_t       aNameSize);
```

Comment:
```suggestion
                                            char (&aName)[Name::kMaxNameSize]);
```
Can we still use a pointer to a fixed size array?

Reply:
Since this function will be used by OpenThread c apis, I believe it's no longer possible.